zellij-server/src/route.rs

```diff
@@ -26,7 +27,7 @@ pub(crate) fn route_action(
     _os_input: &dyn ServerOsApi,
     to_server: &SenderWithContext<ServerInstruction>,
     client_id: ClientId,
-) -> bool {
+) -> Result<bool> {
     let mut should_break = false;
 
```

Comment:
Sorry for not raising this earlier... I think we should add an error context to all the `?` inside this function. I missed that this actually *isn't* the threads main function, but only called from `route_thread_main`. I'm fine with sticking to `?` in `route_thread_main` because we will call `.fatal()` on it in the next step and it's totally clear where the error comes from and what we tried to do.

I think an error context like this:

```suggestion
let err_context = || format!("failed to route action for client {client_id}: {action:?}");
```

should do, and then you replace all `?` in this function with `.with_context(err_context)?`. Is that alright for you?

Reply:
It seems that putting the `action` in the `err_context` will report an error: cannot move out of `action.0` because it is borrowed [E0505] move out of `action.0` occurs

Reply:
That's true, my bad! Then maybe this:

```rust
let err_context = || format!("failed to route action for client {client_id}");
```

Reply:
Thanks for your patience, I'll do that.